src/Polly/Retry/Settings/DelegateOnRetryCallback.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+
+namespace Polly.Retry.Settings;
+
+public class DelegateOnRetryCallback : IOnRetryCallback
```

Comment:
This is a backward compatibility version of all the delegates around. Of course it does nothing to public API surface area, because it simply mimics all the possibilities already given by the API. The complexity is due to forwarding all calls to a single constructor. However, it is an intermediate step to make it possible to introduce the Options/Settings concept without breaking the API. This would be a change to the minor version.